RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractZoneIdentifier.java

```diff
@@ -0,0 +1,398 @@
+            LOG.log(Level.SEVERE, "Unable to find zone identifier files, exception thrown. ", ex);
+        }
+
+        if (zoneFiles == null || zoneFiles.isEmpty()) {
+            return;
+        }
+        
+        ArrayList<Long> knownPathIDs = null;
+        try {
+            knownPathIDs = getPathIDsForType(BlackboardArtifact.ARTIFACT_TYPE.TSK_WEB_DOWNLOAD);
+        } catch (TskCoreException ex) {
+            addErrorMessage(Bundle.ExtractZone_process_errMsg());
+            LOG.log(Level.SEVERE, "Failed to build PathIDs List for TSK_WEB_DOWNLOAD", ex);
+        }
+
+        if (knownPathIDs == null) {
+            return;
+        }
+
+        Collection<BlackboardArtifact> sourceArtifacts = new ArrayList<>();
+        Collection<BlackboardArtifact> downloadArtifacts = new ArrayList<>();
+
+        for (AbstractFile zoneFile : zoneFiles) {
+            try {
+                processZoneFile(context, dataSource, zoneFile, sourceArtifacts, downloadArtifacts);
+            } catch (TskCoreException ex) {
+                addErrorMessage(Bundle.ExtractZone_process_errMsg());
+                String message = String.format("Failed to process zone identifier file  %s", zoneFile.getName()); //NON-NLS
+                LOG.log(Level.WARNING, message, ex);
+            }
+        }
+
+        IngestServices services = IngestServices.getInstance();
+
+        if (!sourceArtifacts.isEmpty()) {
+            services.fireModuleDataEvent(new ModuleDataEvent(
+                    RecentActivityExtracterModuleFactory.getModuleName(),
+                    BlackboardArtifact.ARTIFACT_TYPE.TSK_DOWNLOAD_SOURCE, sourceArtifacts));
+        }
+
+        if (!downloadArtifacts.isEmpty()) {
+            services.fireModuleDataEvent(new ModuleDataEvent(
+                    RecentActivityExtracterModuleFactory.getModuleName(),
+                    BlackboardArtifact.ARTIFACT_TYPE.TSK_WEB_DOWNLOAD, downloadArtifacts));
+        }
+    }
+
+    /**
+     * Process a single Zone Identifier file.
+     *
+     * @param context           IngetJobContext
+     * @param dataSource        Content
+     * @param zoneFile          Zone Indentifier file
+     * @param sourceArtifacts   List for TSK_DOWNLOAD_SOURCE artifacts
+     * @param downloadArtifacts List for TSK_WEB_DOWNLOAD aritfacts
+     */
+    private void processZoneFile(IngestJobContext context, Content dataSource,
+            AbstractFile zoneFile, Collection<BlackboardArtifact> sourceArtifacts,
+            Collection<BlackboardArtifact> downloadArtifacts) throws TskCoreException {
+
+        ZoneIdentifierInfo zoneInfo = null;
+
+        try {
+            zoneInfo = new ZoneIdentifierInfo(zoneFile);
+        } catch (IOException ex) {
+            String message = String.format("Unable to parse temporary File for %s", zoneFile.getName()); //NON-NLS
+            LOG.log(Level.WARNING, message, ex);
+        }
+
+        if (zoneInfo == null) {
+            return;
```

Comment:
again see comment about putting the return in the catch block.  This is not a functional thing just stylistic so maybe @bcarrier or @rcordovano can way in on what they prefer.

Reply:
I do prefer to make it clear what's going on, I am afraid if someone was to change the code so that null was a possibility, is return would be lost in the catch causing null pointer exceptions later.